msal/src/Microsoft.Identity.Client/CacheV2/Impl/PathStorageWorker.cs

```diff
@@ -0,0 +1,551 @@
+            {
+                _fileIo.DeleteFile(
+                    _credentialPathManager.GetCredentialPath(
+                        homeAccountId,
+                        environment,
+                        string.Empty,
+                        string.Empty,
+                        familyId,
+                        CredentialType.OAuth2RefreshToken));
+            }
+
+            if (!string.IsNullOrWhiteSpace(clientId))
+            {
+                _fileIo.DeleteFile(
+                    _credentialPathManager.GetCredentialPath(
+                        homeAccountId,
+                        environment,
+                        string.Empty,
+                        clientId,
+                        string.Empty,
+                        CredentialType.OAuth2RefreshToken));
+            }
+        }
+
+        private void DeleteAccessTokens(
+            string homeAccountId,
+            string environment,
+            string realm,
+            string clientId,
+            string target)
+        {
+            string credentialPath = _credentialPathManager.GetCredentialPath(
+                homeAccountId,
+                environment,
+                realm,
+                clientId,
+                string.Empty,
+                CredentialType.OAuth2AccessToken);
+            bool isFileEmpty = false;
+            ReadModifyWrite(
+                credentialPath,
+                fileContentJson =>
+                {
+                    RemoveAccessTokenWithScopes(fileContentJson, target);
+                    if (!fileContentJson.HasValues)
+                    {
+                        isFileEmpty = true;
+                    }
+
+                    return fileContentJson;
+                });
+
+            if (isFileEmpty)
+            {
+                _fileIo.DeleteFile(credentialPath);
+            }
+        }
+
+        public void RemoveAccessTokenWithScopes(JObject accessTokens, string target)
+        {
+            KeyValuePair<string, JToken>? kvp = FindAccessTokenIterWithScopes(accessTokens, target);
+            if (kvp.HasValue)
+            {
+                accessTokens.Remove(kvp.Value.Key);
+            }
+        }
+
+        public void ReadModifyWrite(string relativePath, Func<JObject, JObject> modify)
+        {
+            _fileIo.ReadModifyWrite(
+                relativePath,
+                existingBytes =>
+                {
+                    var fileContentJson = DecryptParse(existingBytes, relativePath);
+                    fileContentJson = modify(fileContentJson);
+                    string serializedJson = fileContentJson.ToString();
+                    return Encrypt(serializedJson);
+                });
+        }
+
+        public byte[] Encrypt(string input)
+        {
+            return Encoding.UTF8.GetBytes(input);
+        }
+
+        private byte[] Decrypt(byte[] input)
+        {
+            return input;
+        }
```

Comment:
assuming there should be a feature flag or similar to flip encryption on/off.

Reply:
added TODO block here